src/main/kotlin/com/exactpro/th2/codec/moldudp64/MoldUdp64Codec.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.codec.moldudp64
+
+import com.exactpro.th2.codec.api.IPipelineCodec
+import com.exactpro.th2.codec.moldudp64.MoldUdp64CodecFactory.Companion.PROTOCOL
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.MessageGroup
+import com.exactpro.th2.common.message.messageType
+
+typealias MessageName = String
+typealias MessageType = String
+
+class MoldUdp64Codec(private val settings: MoldUdp64CodecSettings) : IPipelineCodec {
+    override fun encode(messageGroup: MessageGroup): MessageGroup {
+        val messages = messageGroup.messagesList
+
+        if (messages.isEmpty()) return messageGroup
+
+        require(messages[0].hasMessage()) { "Header message must be a parsed message" }
```

Comment:
I am not sure that the header message is required...maybe we should take default values for Session and Sequence Number parameters. The connect box will replace them anyway